go-controller/pkg/clustermanager/routeadvertisements/controller.go

```diff
@@ -490,14 +498,24 @@ func (c *Controller) generateFRRConfigurations(ra *ratypes.RouteAdvertisements)
 		return eipsByNodesByNetworks[nodeName], nil
 	}
 
-	// helper to gather host subnets and egress ips as prefixes
-	getPrefixes := func(nodeName string, network string) ([]string, error) {
+	// helper to gather the following prefixes:
+	//  - EgressIPs
+	//  - host subnets for networks with networkTopology layer3
+	//  - network subnets for networks with networkTopology layer2
+	getPrefixes := func(nodeName, network, networkTopology string, networkSubnets []string) ([]string, error) {
 		// gather host subnets
 		var subnets []string
 		if advertisements.Has(ratypes.PodNetwork) {
-			subnets, err = getHostSubnets(nodeName, network)
-			if err != nil || len(subnets) == 0 {
-				return nil, fmt.Errorf("%w: will wait for subnet annotation to be set for node %q and network %q: %w", errConfig, nodeName, network, err)
+			if networkTopology == types.Layer2Topology {
+				subnets = networkSubnets
+				if len(subnets) == 0 {
+					return nil, fmt.Errorf("%w: no layer2 subnets found", errConfig)
+				}
+			} else {
+				subnets, err = getHostSubnets(nodeName, network)
+				if err != nil || len(subnets) == 0 {
+					return nil, fmt.Errorf("%w: will wait for subnet annotation to be set for node %q and network %q: %w", errConfig, nodeName, network, err)
+				}
```

Comment:
I guess I was expecting something else and I probably didn't express myself well enough.
`selectedNetworks.hostNetworkSubnets` can already be prepopulated for L2 networks so we wouldn't really need to call `getPrefixes`.

Reply:
Scratch that. I guess this is true but would require some level of shifting to also know the node names. Had I known this I probably would not have asked for any change from your original proposal. Sorry. What do you want to do?

Reply:
I would just keep the current approach if that is ok with you.